File: tests/TimedTest/cpp/Sensor.cc
#include <Sensor.h>

namespace timedtest {
   namespace components {
      
      Sensor::Sensor(_component_conf &config, riaps::Actor &actor) :
      SensorBase(config, actor) {
      }
      
      void Sensor::OnClock(riaps::ports::PortBase *port) {
          capnp::MallocMessageBuilder builder;
          auto msg = builder.initRoot<DummyT>();
          msg.setMsg("0000");
          SendReady(builder, msg);
      }
      
      
      void Sensor::OnRequest(const DummyT::Reader &message,
      riaps::ports::PortBase *port)
      {
          if (port->GetPortBaseConfig()->isTimed) {
              _logger->info("{}, sendTs: {}.{}, recvTs: {}.{}", __FUNCTION__,
                            port->AsRecvPort()->GetLastSendTimestamp().tv_sec,
                            port->AsRecvPort()->GetLastSendTimestamp().tv_nsec,
                            port->AsRecvPort()->GetLastRecvTimestamp().tv_sec,
                            port->AsRecvPort()->GetLastRecvTimestamp().tv_nsec);
          } else {
              _logger->info("{}", __FUNCTION__);
          }

          capnp::MallocMessageBuilder builder;
          auto msg = builder.initRoot<DummyT>();
          msg.setMsg("FromSensor");
          SendRequest(builder, msg);
      }
      
      void Sensor::OnGroupMessage(const riaps::groups::GroupId& groupId,
      capnp::FlatArrayMessageReader& capnpreader, riaps::ports::PortBase* port){
         
      }
      
      Sensor::~Sensor() {
         
      }
   }
}

riaps::ComponentBase *create_component(_component_conf &config, riaps::Actor &actor) {
   auto result = new timedtest::components::Sensor(config, actor);
   return result;
}

void destroy_component(riaps::ComponentBase *comp) {
   delete comp;
}

File: tests/TimedTest/python/Sensor.py
#Sensor.py
from riaps.run.comp import Component
import os
import logging

class Sensor(Component):
    def __init__(self):
        super(Sensor, self).__init__()	        
        self.pid = os.getpid()
        self.logger.info("(PID %s) - starting Sensor",str(self.pid))
        

    def on_clock(self):
        now = self.clock.recv_pyobj()
        self.logger.info('PID(%s) - on_clock(): %s',str(self.pid),str(now))

    def on_request(self):
        msg = self.request.recv_pyobj()
        self.logger.info("PID (%s) - on_query():%s",str(self.pid),str(msg))
    
    def __destroy__(self):			
        self.logger.info("(PID %s) - stopping Sensor",str(self.pid))   	        	        
